From cbbe1d2d3a4901dc9ff33fabf80ef7954d08194f Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Sat, 4 Jan 2020 10:32:23 +0100
Subject: [PATCH] Enable legacy adaptive playback for QCOM_BSP_LEGACY

Adaptive playback got broken in Q, and this is needed
to fix issues with video playback in apps like YouTube etc.

Change-Id: Ieb440b93222e67043eb5ceda50c0b5eea3198695
(cherry picked from commit 6c0869071db4a89787ff497e838bcc9415fd0904)
(cherry picked from commit 44cd6b9e72b906973e4582e9f402cff647b4736d)
---
 media/libstagefright/omx/Android.bp          | 1 +
 media/libstagefright/omx/OMXNodeInstance.cpp | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/media/libstagefright/omx/Android.bp b/media/libstagefright/omx/Android.bp
index 54c5697c14..d9e386e8d8 100644
--- a/media/libstagefright/omx/Android.bp
+++ b/media/libstagefright/omx/Android.bp
@@ -19,6 +19,7 @@ license {
 
 cc_library_shared {
     name: "libstagefright_omx",
+    defaults: ["stagefright_qcom_legacy_defaults"],
     vendor_available: true,
     vndk: {
         enabled: true,
diff --git a/media/libstagefright/omx/OMXNodeInstance.cpp b/media/libstagefright/omx/OMXNodeInstance.cpp
index 44c54e4754..008ae53ba3 100644
--- a/media/libstagefright/omx/OMXNodeInstance.cpp
+++ b/media/libstagefright/omx/OMXNodeInstance.cpp
@@ -453,7 +453,11 @@ OMXNodeInstance::OMXNodeInstance(
     mGraphicBufferEnabled[0] = false;
     mGraphicBufferEnabled[1] = false;
     mIsSecure = AString(name).endsWith(".secure");
+#ifdef QCOM_BSP_LEGACY
+    mLegacyAdaptiveExperiment = true;
+#else
     mLegacyAdaptiveExperiment = ADebug::isExperimentEnabled("legacy-adaptive");
+#endif
 }
 
 OMXNodeInstance::~OMXNodeInstance() {
